enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,467 @@
+such as the ExternalDNS TTL annotation.
+
+Currently, the Ingress Operator publishes wildcard DNS Records to point an Ingress Controller's domain to the Ingress Controller's
+LoadBalancer type Service (or alternatively, a cluster administrator configures this for their administrator-managed Load Balancer, etc).
+In the future, the Ingress Operator could create DNSRecord Custom Resources to be managed by ExternalDNS (instead of the Ingress Operator's bespoke
+DNS provider). The Ingress Controller API could be expanded to include the ability for a cluster administrator to specify whether or not a Wildcard DNSRecord
+is desired.
+
+
+Here is an example of what an ExternalDNS deployment (created by the operator) would look like:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: external-dns
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      serviceAccountName: external-dns
+      containers:
+      - name: external-dns-private
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        - --policy=upsert-only # would prevent ExternalDNS from deleting any records, omit to enable full synchronization
+        # can only have one zone at a time if domains are overlapping over zones
+        - --aws-zone-type=private
+        - --registry=txt
+        - --txt-owner-id=externalDNS-operator
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7979
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          # Base64 decode from cloud-credentials secret generated by cloud credentials request
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+      - name: external-dns-public
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
+        - --policy=upsert-only # would prevent ExternalDNS from deleting any records, omit to enable full synchronization
+        # can only have one zone at a time if domains are overlapping over zones
+        - --aws-zone-type=public
+        - --registry=txt
+        - --txt-owner-id=my-identifier
+        - --log-level=debug
+        # Needs to be unique per container
+        - --metrics-address=:7980
+        # Filter from aws dev account
+        - --zone-id-filter=/hostedzone/myzone
+        env:
+          # Base64 decode from cloud-credentials secret generated by cloud credentials request
+          - name: AWS_ACCESS_KEY_ID
+            value: abc
+          - name: AWS_SECRET_ACCESS_KEY
+            value: xyz
+```
+
+Note the use of multiple ExternalDNS containers within the same pod int he above deployment. Metrics listening ports would have
```

Comment:
s/he/the/

Reply:
Have you considered the pros/cons of running the pub/private instances of ext-dns as separate deployments instead of separate containers?

Reply:
I have, and other than the fact that metric ports will collide, I can't really think of any cons of running multiple ExternalDNS instances in the same deployment. Did you have any particular downsides in mind?